yt_dlp/extractor/ichinanalive.py

```diff
@@ -157,3 +166,45 @@ def _real_extract(self, url):
             'description': view_data.get('caption'),
             'upload_date': unified_strdate(str_or_none(view_data.get('createdAt'))),
         }
+
+
+class IchinanaLiveVODIE(InfoExtractor):
+    IE_NAME = '17live:vod'
+    _VALID_URL = r'https?://(?:www\.)?17\.live/ja/vod/[^/]+/(?P<video_id>[^/]+)'
```

Comment:
- Need to use `[^/?#]` instead of `[^/]` or else we'll match the URL query/fragment in the ID
- Whenever possible use `id` as the name of the capture group; then the preliminary check done for `--download-archive` is effective
```suggestion
    _VALID_URL = r'https?://(?:www\.)?17\.live/ja/vod/[^/?#]+/(?P<id>[^/?#]+)'
```